### PR TITLE
doc changes

### PR DESCRIPTION
**Pull request by zalmane**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -35,18 +35,73 @@ cd hello_world
...
-{"greeting": "Hello Mr. Bill Adams"}
+{"id": "1", "fname": "john", "lname": "doe", "credit_card": "1234-1234-1234-1234", "country_code": "972", "country_name": "israel", "gender": "M", "full_name": "John Doe", "greeting": "HelMr. John Doe"}
+{"id": "2", "fname": "jane", "lname": "doe", "credit_card": "1000-2000-3000-4000", "country_code": "972", "country_name": "israel", "gender": "F", "full_name": "Jane Doe", "greeting": "HelMs. Jane Doe"}
+{"id": "3", "fname": "bill", "lname": "adams", "credit_card": "9999-8888-7777-666", "country_code": "1", "country_name": "usa", "gender": "M", "full_name": "Bill Adams", "greeting": "HelMr. Bill Adams"}
 ```
 
-That's it! You've created your first job that loads data from CSV, runs it through a series of transformation steps, and shows the data to the standard output. A good start. Read on for a more detailed tutorial or check out the [reference](reference/blocks.md) to see the different block types currently available.
+That's it! You've created your first job that loads data from CSV, runs it through a series of transformation steps, and shows the data to the standard output. A good start!
+
+## Enhance The Job Definition
```
Comment (spicy-sauce) [inline docs/index.md]: The --> the

File: docs/index.md [hunk 2]
```diff
@@ -35,18 +35,73 @@ cd hello_world
...
 
-Let's run our first job. It is pre-defined in the samples folder as part of the `init` command:
+Let's run our first job. It is pre-defined in the samples folder as part of the `init` command. Our job is going to load a CSV file from the samples folder and transform it into JSON:
 
 ```bash
 datayoga run sample.hello
 ```
 
-If all goes well, you should see some startup logs, and eventually:
+If all goes well, you should see a list of JSON values that correspond to each line in the input csv file:
```
Comment (spicy-sauce) [inline docs/index.md]: csv --> CSV

File: docs/index.md [hunk 3]
```diff
@@ -35,18 +35,73 @@ cd hello_world
...
 
 ```yaml
-{"greeting": "Hello Mr. John Doe"}
-{"greeting": "Hello Ms. Jane Doe"}
-{"greeting": "Hello Mr. Bill Adams"}
+{"id": "1", "fname": "john", "lname": "doe", "credit_card": "1234-1234-1234-1234", "country_code": "972", "country_name": "israel", "gender": "M", "full_name": "John Doe", "greeting": "HelMr. John Doe"}
+{"id": "2", "fname": "jane", "lname": "doe", "credit_card": "1000-2000-3000-4000", "country_code": "972", "country_name": "israel", "gender": "F", "full_name": "Jane Doe", "greeting": "HelMs. Jane Doe"}
+{"id": "3", "fname": "bill", "lname": "adams", "credit_card": "9999-8888-7777-666", "country_code": "1", "country_name": "usa", "gender": "M", "full_name": "Bill Adams", "greeting": "HelMr. Bill Adams"}
```
Comment (spicy-sauce) [inline docs/index.md]: `HelMr.` and `HelMs.`, is that correct?

File: docs/index.md [hunk 4]
```diff
@@ -35,18 +35,73 @@ cd hello_world
...
+That's it! You've created your first job that loads data from CSV, runs it through a series of transformation steps, and shows the data to the standard output. A good start!
+
+## Enhance The Job Definition
+
+Let's customize the structure of the resulting JSON.
+The jobs are located under the `jobs` folder and can be arrange into modules. The `sample.hello` job you just ran is located in `jobs/sample/hello.yaml`.
+
+Add a new step of type `map` into the chain of steps. To do this, open the job definition in a text editor and add the section highlighted in bold to the job definition:
+
+<pre><code>
```
Comment (spicy-sauce) [inline docs/index.md:62]: this vanishes the YAML highlighting, what about keeping it (```yaml) and add a comment that stresses out the new section?

File: docs/index.md [hunk 5]
```diff
@@ -35,18 +35,73 @@ cd hello_world
...
+          language: jmespath
+          expression: concat([capitalize(fname), ' ' , capitalize(lname)])
+        - field: greeting
+          language: sql
+          expression: "'Hello ' || CASE WHEN gender = 'F' THEN 'Ms.' WHEN gender = 'M' THEN 'Mr.' ELSE 'N/A' END || ' ' || full_name"
+<b>
+  - uses: map
+    with:
+      expression: |
+        { greeting: greeting,
```
Comment (spicy-sauce) [inline docs/index.md:81]: format it, so this will be in the next line

File: docs/index.md [hunk 6]
```diff
@@ -35,18 +35,73 @@ cd hello_world
...
+```bash
+datayoga run sample.hello
+```
+
+You should now see the modified JSON as the output:
+
+```bash
+{"greeting": "HelMr. John Doe", "details": {"first_name": "john", "last_name": "doe"}}
+{"greeting": "HelMs. Jane Doe", "details": {"first_name": "jane", "last_name": "doe"}}
+{"greeting": "HelMr. Bill Adams", "details": {"first_name": "bill", "last_name": "adams"}}
```
Comment (spicy-sauce) [inline docs/index.md]: `greeting` - same as the previous comment, is this what we should get?

File: docs/introduction.md [hunk 7]
```diff
@@ -6,6 +6,8 @@ nav_order: 1
 
 DataYoga is a framework for building and running streaming or batch data pipelines. The DataYoga CLI helps define and deploy data pipelines using a declarative markup language using YAML files.
 
+![DataYoga overview](images/datayoga_overview.png "Datayoga Overview")
```
Comment (spicy-sauce) [inline docs/introduction.md]: DataYoga overview --> DataYoga Overview
Datayoga Overview -> DataYoga Overview

File: docs/index.md [hunk 8]
```diff
@@ -35,18 +35,73 @@ cd hello_world
...
+{"id": "2", "fname": "jane", "lname": "doe", "credit_card": "1000-2000-3000-4000", "country_code": "972", "country_name": "israel", "gender": "F", "full_name": "Jane Doe", "greeting": "HelMs. Jane Doe"}
+{"id": "3", "fname": "bill", "lname": "adams", "credit_card": "9999-8888-7777-666", "country_code": "1", "country_name": "usa", "gender": "M", "full_name": "Bill Adams", "greeting": "HelMr. Bill Adams"}
 ```
 
-That's it! You've created your first job that loads data from CSV, runs it through a series of transformation steps, and shows the data to the standard output. A good start. Read on for a more detailed tutorial or check out the [reference](reference/blocks.md) to see the different block types currently available.
+That's it! You've created your first job that loads data from CSV, runs it through a series of transformation steps, and shows the data to the standard output. A good start!
+
+## Enhance The Job Definition
+
+Let's customize the structure of the resulting JSON.
```
Comment (spicy-sauce) [inline docs/index.md:56]: maybe add a blank line to have those lines separated.

**Review by spicy-sauce — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 1 above)
Comment (zalmane) [reply docs/index.md]: done

File: docs/index.md (hunk 2 above)
Comment (zalmane) [reply docs/index.md]: done

File: docs/index.md (hunk 4 above)
Comment (zalmane) [reply docs/index.md:62]: will be hard to follow I think. Even this is not ideal. would be nicer if it was highlighted somehow...

File: docs/index.md (hunk 5 above)
Comment (zalmane) [reply docs/index.md:81]: I will open a bug. apparently this is defined as an object and not string. so if you try to put a | sign it fails. we are better off parsing this in the code I think and leaving the yaml validation as string

File: docs/index.md (hunk 6 above)
Comment (zalmane) [reply docs/index.md]: fixed

File: docs/index.md (hunk 3 above)
Comment (zalmane) [reply docs/index.md]: fixed

File: docs/introduction.md (hunk 7 above)
Comment (zalmane) [reply docs/introduction.md]: done

File: docs/index.md (hunk 4 above)
Comment (spicy-sauce) [reply docs/index.md:62]: OK

**Review by spicy-sauce — APPROVED** [review]
